frontend/apps/core-app/src/utils/ReactHookFormTransformer.ts

```diff
@@ -18,16 +41,50 @@ export const toReactHookForm = (
   }, {});
 
 export const fromReactHookForm = (
-  originalData: FormWithRecord<Recipient>[],
-  value: FieldValue<any>,
-): FormWithRecord<Recipient>[] =>
+  originalData: FormWithRecord<Record>[],
+  value: RHFFieldValue<any>,
+): FormWithRecord<Record>[] =>
   originalData.map((datum) => ({
     form: datum.form,
     record: {
       ...datum.record,
-      values: datum.form.fields.map((field) => ({
-        fieldId: field.id,
-        value: value[datum.form.id][field.id],
-      })),
+      values: datum.form.fields.map((field) => {
+        if (getFieldKind(field.fieldType) === FieldKind.SubForm) {
+          const v = value[datum.form.id][field.id].map(
+            (vv) =>
+              fromReactHookForm(
+                [
+                  {
+                    form: {
+                      id: field.id,
+                      databaseId: datum.form.databaseId,
+                      folderId: datum.form.folderId,
+                      name: '',
```

Comment:
do we not need a name here? isn't formname required?

Reply:
It's required, but not used by subforms